railgun/test/integration/knative_test.go

```diff
@@ -0,0 +1,286 @@
+//+build integration_tests
+
+package integration
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"time"
+
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	types "k8s.io/apimachinery/pkg/types"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/kind"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	knativeCrds = "https://github.com/knative/serving/releases/download/v0.13.0/serving-crds.yaml"
+	knativeCore = "https://github.com/knative/serving/releases/download/v0.13.0/serving-core.yaml"
+)
+
+func isKnativeReady(ctx context.Context, cluster kind.Cluster) bool {
+	timeout := 1
+	for timeout < 60 {
+		podList, err := cluster.Client().CoreV1().Pods("knative-serving").List(ctx, metav1.ListOptions{})
+		if err != nil {
+			fmt.Println("failed retrieving knative pods.", err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+
+		ready := true
+		if len(podList.Items) != 4 {
+			fmt.Println("expected 4 pods, found ", len(podList.Items))
+			time.Sleep(1 * time.Second)
+			continue
+		}
+
+		for _, pod := range podList.Items {
+			if pod.Status.Phase != v1.PodRunning {
+				ready = false
+			}
+		}
+		if ready == true {
+			fmt.Println("All knative pods are up and ready.")
+			fmt.Println("Covering a window that webhook has been configured itself but not ready to receive traffic yet.")
+			time.Sleep(3 * time.Second)
+			return true
+		}
+		if ready == false {
+			time.Sleep(1 * time.Second)
+		}
+	}
+
+	fmt.Println("Failed to bring up knative resources within 60 seconds.")
```

Comment:
It seems this entire function is meant to wait for knative resources to be ready, but creates it's own machinery for doing so. It seems to me that we should use `require.Eventually` for this purpose instead as this will be more consistent with the rest of the integration tests:

https://pkg.go.dev/github.com/stretchr/testify@v1.7.0/require#Eventually